RadixWallet/Core/FeaturePrelude/FactorSourceCard/FactorSourceCard.swift

```diff
@@ -0,0 +1,295 @@
+
+			if accounts.count > 0, personas.count > 0 {
+				return Card.linkedAccountsAndPersonas(accountsString, personasString)
+			} else if accounts.count > 0 {
+				return Card.linkedAccountsOrPersonas(accountsString)
+			} else if personas.count > 0 {
+				return Card.linkedAccountsOrPersonas(personasString)
+			}
+			return ""
+		}
+	}
+}
+
+extension FactorSourceCard {
+	init?(
+		kind: Kind,
+		mode: Mode,
+		messages: [FactorSourceCardDataSource.Message] = [],
+		isExpanded: Bool = false,
+		onRemoveTapped: (() -> Void)? = nil
+	) {
+		guard
+			let icon = kind.factorSourceKind.icon,
+			let title = kind.title
+		else { return nil }
+
+		switch kind {
+		case let .genericDescription(factorSourceKind):
+			guard let details = factorSourceKind.details else { return nil }
+
+			self = .init(
+				kind: kind,
+				mode: mode,
+				dataSource: .init(
+					icon: icon,
+					title: title,
+					subtitle: details,
+					messages: messages
+				),
+				isExpanded: isExpanded,
+				onRemoveTapped: onRemoveTapped
+			)
+		case let .instanceCompact(factorSource):
+			self = .init(
+				kind: kind,
+				mode: mode,
+				dataSource: .init(
+					icon: icon,
+					title: factorSource.name,
+					messages: messages
+				),
+				isExpanded: isExpanded,
+				onRemoveTapped: onRemoveTapped
+			)
+		case let .instanceRegular(factorSource):
+			guard let details = factorSource.factorSourceKind.details else { return nil }
+
+			self = .init(
+				kind: kind,
+				mode: mode,
+				dataSource: .init(
+					icon: icon,
+					title: factorSource.name,
+					subtitle: details,
+					messages: messages
+				),
+				isExpanded: isExpanded,
+				onRemoveTapped: onRemoveTapped
+			)
+		case let .instanceLastUsed(factorSource, accounts, personas):
+			self = .init(
+				kind: kind,
+				mode: mode,
+				dataSource: .init(
+					icon: icon,
+					title: factorSource.name,
+					lastUsedOn: factorSource.common.lastUsedOn,
+					messages: messages,
+					accounts: accounts,
+					personas: personas
+				),
+				isExpanded: isExpanded,
+				onRemoveTapped: onRemoveTapped
+			)
+		}
+	}
+}
+
+extension FactorSourceCard {
+	enum Kind {
+		case genericDescription(FactorSourceKind)
+		case instanceCompact(factorSource: FactorSource)
+		case instanceRegular(factorSource: FactorSource)
+		case instanceLastUsed(
+			factorSource: FactorSource,
+			accounts: [Account] = [],
+			personas: [Persona] = []
+		)
+	}
+
+	enum Mode {
+		case display
+		case selection(type: SelectionType, isSelected: Bool)
+		case removal
+	}
+
+	enum SelectionType {
+		case radioButton
+		case checkmark
+	}
+}
+
+extension FactorSourceCard.Kind {
+	var factorSourceKind: FactorSourceKind {
+		switch self {
+		case let .genericDescription(factorSourceKind):
+			factorSourceKind
+		case let .instanceCompact(factorSource),
+		     let .instanceRegular(factorSource),
+		     let .instanceLastUsed(factorSource, _, _):
+			factorSource.factorSourceKind
```

Comment:
this would change to
```suggestion
		case let .instanceCompact(factorSource, _)
			factorSource.factorSourceKind
```